File: tests/e2e/cases/09-type-coverage/max_i16/max_i16_verified.cpp
/*
DEMO: max of two i16 values.

Returns the larger of `a` and `b`. Exercises:
  * `int16_t` parameter + return — width not seen by the existing
    integer suites.
  * Signed-greater-than at 16 bits (lowers to `icmp sgt i16`).

Expected verdict: VERIFIED.
*/

#include <cstdint>

int16_t max_i16(int16_t a, int16_t b) {
    return a > b ? a : b;
}

File: tests/e2e/cases/09-type-coverage/max_i16/max_i16_verified.rs
// DEMO: max of two i16 values (Rust mirror of max_i16_verified.cpp).
// Exercises i16 parameter + return + signed-compare, absent from earlier suites.

fn max_i16(a: i16, b: i16) -> i16 {
    if a > b { a } else { b }
}
